### PR TITLE
scylla-cql: Use `#[cfg(feature = "time")]` for `time` related `impl SerializeCql`'s

### PR DESCRIPTION
**Pull request by oeb25**
Fixes: #897, see the issue for details.

Additionally I went ahead and checked that `--no-default-features -F time` also works.

## Pre-review checklist

- [X] I have split my patch into logically separate commits.
- [X] All commit messages clearly explain what they change and why.
- [ ] I added relevant tests for new features and bug fixes.
- [X] All commits compile, pass static checks and pass test.
- [X] PR description sums up the changes and reasons why they should be introduced.
- [ ] I have provided docstrings for the public items that I want to introduce.
- [ ] I have adjusted the documentation in `./docs/source/`.
- [ ] I added appropriate `Fixes:` annotations to PR description.

### CODE REVIEW

**Review by piodul — COMMENTED** [review]
Thanks.

When I reviewed the changes which introduced the `time` and `chrono` features, I was a bit too optimistic and thought that only checking compilation with and without all features would suffice. Would you mind adding some steps to `.github/workflows.yml` to check that everything compiles when only one of the `time`, `chrono`, `secret` features is enabled? You could base them on the existing `Cargo check with all serialization features` step.

**Comment by oeb25** [comment]
Good idea! I've added three steps to the job, and it seems to work.

As a complete side note, I've found that [`- uses: Swatinem/rust-cache@v2`](https://github.com/Swatinem/rust-cache) can speed-up CI quite a bit! So if that's something that you are interested in I can open up a PR to measure the difference?

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -34,6 +34,12 @@ jobs:
       run: cargo check --all-targets --manifest-path "scylla/Cargo.toml" --features ""
     - name: Cargo check with all serialization features
       run: cargo check --all-targets --manifest-path "scylla/Cargo.toml" --features "full-serialization"
+    - name: Cargo check with all secret feature
+      run: cargo check --all-targets --manifest-path "scylla/Cargo.toml" --features "secret"
+    - name: Cargo check with all chrono feature
+      run: cargo check --all-targets --manifest-path "scylla/Cargo.toml" --features "chrono"
+    - name: Cargo check with all time feature
```
Comment (Lorak-mmk) [inline .github/workflows/rust.yml:41]: Please remove "all" words from descriptions.

**Comment by Lorak-mmk** [comment]
> Good idea! I've added three steps to the job, and it seems to work.
> 
> As a complete side note, I've found that [`- uses: Swatinem/rust-cache@v2`](https://github.com/Swatinem/rust-cache) can speed-up CI quite a bit! So if that's something that you are interested in I can open up a PR to measure the difference?

Sure, that would be great

File: .github/workflows/rust.yml (hunk 1 above)
Comment (oeb25) [reply .github/workflows/rust.yml]: Oops!

**Comment by Lorak-mmk** [comment]
Code change looks good to me now. Could you squash 2 commits that modify CI?

**Comment by oeb25** [comment]
Squashed!

**Review by Lorak-mmk — APPROVED** [review]
Looks good to me, @piodul ?

**Review by piodul — APPROVED** [review]

**Pull request merged**